src/burn/engine/apply.cpp

```diff
@@ -452,6 +454,9 @@ extern "C" HRESULT ApplyCache(
     hr = UserExperienceInterpretExecuteResult(pUX, FALSE, MB_OKCANCEL, nResult);
     ExitOnRootFailure(hr, "UX aborted cache.");
 
+    // When in cache mode, we want to download in the background (if the download mechanism supports it).
+    BOOL fPreferBackgroundDownload = (BOOTSTRAPPER_ACTION_CACHE == pPlan->action);
```

Comment:
This seems pretty magical. Seems like BITS needs more callbacks (or something) to handle it's priority. Not great answer here but this may be the best can do in v3.x. @barnson, thoughts?